cmd/booster-bitswap/blockfilter/blockfilter.go

```diff
@@ -0,0 +1,244 @@
+package blockfilter
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/multiformats/go-multibase"
+)
+
+var log = logging.Logger("booster-bitswap")
+
+// BadBitsDenyList is the URL for well known bad bits list
+const BadBitsDenyList string = "https://badbits.dwebops.pub/denylist.json"
+
+// UpdateInterval is the default interval at which the public list is refected and updated
+const UpdateInterval = 5 * time.Minute
+
+// DenyListFetcher is a function that fetches a deny list in the json style of the BadBits list
+// The first return value indicates whether any update has occurred since the last fetch time
+// The second return is a stream of data if an update has occurred
+// The third is any error
+type DenyListFetcher func(lastFetchTime time.Time) (bool, io.ReadCloser, error)
+
+const expectedListGrowth = 128
+
+// FetchBadBitsList is the default function used to get the BadBits list
+func FetchBadBitsList(ifModifiedSince time.Time) (bool, io.ReadCloser, error) {
+	req, err := http.NewRequest("GET", BadBitsDenyList, nil)
+	if err != nil {
+		return false, nil, err
+	}
+	// set the modification sync header, assuming we are not given time zero
+	if !ifModifiedSince.IsZero() {
+		req.Header.Set("If-Modified-Since", ifModifiedSince.Format(http.TimeFormat))
+	}
+	response, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, nil, err
+	}
+	if response.StatusCode == http.StatusNotModified {
+		return false, nil, nil
+	}
+	if response.StatusCode < 200 && response.StatusCode > 299 {
+		return false, nil, errors.New("http status code was not success")
```

Comment:
It could be helpful to include the error code and the error text in the returned error.
I think the error text should be in response.Body.